### ISSUE TITLE
Subsequent reduction operations yield wrong result

### ISSUE THREAD

**Issue by ClaudiaComito** [helmholtz-analytics/heat#228]
Example: a.min(0).min(0), which should be equivalent to a.min((0,1)). Final tensor shape is correct, but tensor is wrong.

```
>>> import heat as ht
>>> a = ht.random.randn(3,5,7)
>>> a
tensor([[[ 0.5370,  0.4091, -0.5320,  1.1930, -1.3061,  1.8513,  0.5728],
         [-0.2762,  0.6790,  1.0925, -1.2878, -0.7464, -1.4269, -0.1732],
         [-0.1401, -1.3958, -0.7422, -0.5591, -0.5358, -0.0032, -0.7736],
         [ 0.5136, -1.5949, -0.1004, -0.5931,  2.0396, -1.0171, -1.3143],
         [ 0.2896,  0.2842, -1.7404, -1.4270, -0.7204,  0.4346, -1.4595]],

        [[-0.0569, -0.3858,  0.5960,  0.2004,  1.2844,  0.7312, -0.7771],
         [ 0.2925, -1.0778,  0.4624, -0.1697, -0.0966, -0.8518,  0.6556],
         [-0.3455,  0.3645,  0.7149, -0.9584, -0.9588,  1.9508, -0.1186],
         [ 0.5864, -0.8102, -0.8512,  0.7835,  1.5131,  0.1037,  0.1493],
         [ 0.4183, -0.9922, -1.1814, -0.6525,  2.8735,  1.8134, -0.1592]],

        [[-0.9499, -1.6196,  0.1174, -0.1777, -1.0976, -1.2244, -0.2953],
         [ 1.1051,  1.0881,  1.2690,  0.5827, -1.2657,  0.3517, -1.6968],
         [ 0.4199,  1.9549,  0.8111, -0.4726, -1.1477,  0.2943,  0.8426],
         [ 0.8430, -0.8488, -0.6329,  1.0462, -0.2560, -0.0703,  0.1082],
         [ 0.6439, -0.2240, -0.8969,  0.9227,  0.2475, -2.5963,  0.0393]]])
>>> b = a.min(0)
>>> b
tensor([[[-0.9499, -1.6196, -0.5320, -0.1777, -1.3061, -1.2244, -0.7771],
         [-0.2762, -1.0778,  0.4624, -1.2878, -1.2657, -1.4269, -1.6968],
         [-0.3455, -1.3958, -0.7422, -0.9584, -1.1477, -0.0032, -0.7736],
         [ 0.5136, -1.5949, -0.8512, -0.5931, -0.2560, -1.0171, -1.3143],
         [ 0.2896, -0.9922, -1.7404, -1.4270, -0.7204, -2.5963, -1.4595]]])
>>> c = b.min(0)
>>> c
tensor([[[-0.9499, -1.6196, -0.5320, -0.1777, -1.3061, -1.2244, -0.7771],
         [-0.2762, -1.0778,  0.4624, -1.2878, -1.2657, -1.4269, -1.6968],
         [-0.3455, -1.3958, -0.7422, -0.9584, -1.1477, -0.0032, -0.7736],
         [ 0.5136, -1.5949, -0.8512, -0.5931, -0.2560, -1.0171, -1.3143],
         [ 0.2896, -0.9922, -1.7404, -1.4270, -0.7204, -2.5963, -1.4595]]])
>>> print(a.shape, b.shape, c.shape)
(3, 5, 7) (5, 7) (7,)
```

By comparison, in PR #220 (not merged to master yet):

```
>>> d = a.min((0,1))
>>> d
tensor([[[-0.9499, -1.6196, -1.7404, -1.4270, -1.3061, -2.5963, -1.6968]]])
>>> d.shape
(7,)
>>> 

```

**Comment by Markus-Goetz** [comment]
That is actually not a bug. This has something to do with the incorrect local shape. In the above example the tensor has a previous shape of (3,5,7), after the first reduction it is (1, 5, 7). A second reduction across axis zero will again result in (1,5,7). Would the first reduction correctly have set the local shape to be (5, 7) you would receive the expected result

**Comment by ClaudiaComito** [comment]
Related to Issue #227 and solved with PR #231.
